docs/dev/dev_setup.rst

```diff
@@ -0,0 +1,217 @@
+make sure they are configured to use
+`strict channel priority <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#>`__
+with the following commands:
+
+.. code-block:: console
+
+    $ conda update conda
+    $ conda config --set channel_priority strict
+
+------------------------------------------------------------------------------
+Fork and Clone the PUDL Repository
+------------------------------------------------------------------------------
+Unless you're part of the Catalyst Cooperative organization already, you'll need to fork
+`the PUDL repository <https://github.com/catalyst-cooperative/pudl>`__
+(see `Forking a repository <https://help.github.com/en/articles/fork-a-repo>`__)
+This makes a copy of it in your personal (or organizational) account on GitHub that
+is independent of, but linked to, the original "upstream" project.
+
+Then, `clone the repository <https://help.github.com/articles/cloning-a-repository/>`__
+from your fork to your local computer where you'll be editing the code or docs.
+This will download the whole history of the project, including the most recent
+version, and put it in a local directory where you can make changes.
+
+-------------------------------------------------------------------------------
+Create the PUDL Development Environment
+-------------------------------------------------------------------------------
+Inside the ``devtools`` directory of your newly cloned repository, you'll see an
+``environment.yml`` file, which specifies the ``pudl-dev`` ``conda`` environment.  You
+can create and activate that environment from within the main repository directory by
+running:
+
+.. code-block:: console
+
+    $ conda update conda
+    $ conda env create --name pudl-dev --file devtools/environment.yml
+    $ conda activate pudl-dev
+
+This environment installs the ``catalystcoop.pudl`` package directly using the code in
+your cloned repository so that it can be edited during development. It also installs
+all of the software PUDL depends on, some packages for testing and quality control,
+working with interactive Jupyter Notebooks, and a few Python packages that have binary
+dependencies which can be easier to satisfy through ``conda`` packages.
+
+-------------------------------------------------------------------------------
+Update the PUDL Development Environment
+-------------------------------------------------------------------------------
+Periodically you will need to update your development software environment. This will
+get you newer versions of existing dependencies, and also incorporate any changes that
+have been made to the environment specification by others. The most reliable way to do
+this is to remove the existing environment and recreate it. From within the top
+directory of the PUDL repository:
+
+.. code-block:: console
+
+    $ git checkout BRANCH_NAME
+    $ git pull
```

Comment:
This part feels like it's lacking an explanation. Is this to ensure that you checkout the branch with the updates to the environment file?

Reply:
Okay, I've tried to make this clearer.